docs/examples/gluster-s3-storage-template/README.md

```diff
@@ -2,111 +2,132 @@
 
 ## Prerequisites
 
-  OpenShift setup is up with master and nodes ready.
+OpenShift setup is up with master and nodes ready.
 
-  cns-deploy tool is ran and heketi service is ready
+cns-deploy tool is ran and heketi service is ready.
 
+## First create a storageclass
 
+We need to create a storageclass which provides access to glusterfs provisioner.
 
+Before creating storageclass, secret needs to be created and labelled. Please note, this step is optional.
 
-## First create a storageclass
+Create the secret:
 
-Create a GlusterFS StorageClass as below, replacing the `rest` parameters with your configuration:
+Replace `NAMESPACE` and `ADMIN_KEY` parameters with your configuration.
 
 ```
-oc create -f ./gluster-s3-storageclass.yaml
+oc create secret generic heketi-${NAMESPACE}-admin-secret --from-literal=key=${ADMIN_KEY} --type=kubernetes.io/glusterfs
 ```
 
-Available at 
-[gluster-s3-storageclass.yaml](./gluster-s3-storageclass.yaml)
+Label the secret:
 
-## Start glusters3 service using template
+Replace `NAMESPACE` parameter with your configuration.
 
 ```
-oc new-app gluster-s3-template.yaml  --param=S3_ACCOUNT=testvolume  --param=S3_USER=adminuser --param=S3_PASSWORD=itsmine --param=VOLUME_CAPACITY=2Gi
+oc label --overwrite secret heketi-${NAMESPACE}-admin-secret glusterfs=s3-heketi-${NAMESPACE}-admin-secret gluster-s3=heketi-${NAMESPACE}-admin-secret
 ```
 
-Note: adjust parameters according to your needs.
+Create a GlusterFS StorageClass as below, replacing the `HEKETI_URL` and `NAMESPACE` parameters with your configuration.
+Also, provide `STORAGE_CLASS` to give a storage class name.
```

Comment:
Combine this sentence into previous sentence. Explain what all the new parameters are for (e.g. `HEKETI_URL` and `STORAGE_CLASS`) and where they'll be used later.

Reply:
Updated. check whether changes looks fine?

Reply:
As they are currently written, all four of these single-sentence lines should be part of the same paragraph with line 38. You could also turn these four lines into a bulleted list if you prefer, but if so reqord the STORAGE_CLASS line to match the other three. :)